controllers/osbuildenvconfig_controller_test.go

```diff
@@ -961,25 +1006,639 @@ var _ = Describe("OSBuildEnvConfig Controller", func() {
+																						Protocol:   "TCP",
+																						TargetPort: intstr.FromInt(22),
+																					},
+																				},
+																				Selector: map[string]string{
+																					"osbuild-worker": internalBuilderName,
+																				},
+																			},
+																		}
+																	})
+
+																	It("Should return an error if failed to get the ssh service of the internal builder", func() {
+																		// given
+																		serviceRepository.EXPECT().Read(requestContext, workerSSHServiceName, operatorNamespace).Return(nil, errFailed)
+																		// when
+																		result, err := reconciler.Reconcile(requestContext, request)
+																		// then
+																		Expect(err).To(Equal(errFailed))
+																		Expect(result).To(Equal(resultRequeue))
+																	})
+
+																	Context("SSH Service for the internal builder not found", func() {
+																		BeforeEach(func() {
+																			serviceRepository.EXPECT().Read(requestContext, workerSSHServiceName, operatorNamespace).Return(nil, errNotFound)
+																		})
+
+																		It("Should return an error if failed to create the SSH Service for the internal builder", func() {
+																			// given
+																			serviceRepository.EXPECT().Create(requestContext, workerSSHService).Return(errFailed)
+																			// when
+																			result, err := reconciler.Reconcile(requestContext, request)
+																			// then
+																			Expect(err).To(Equal(errFailed))
+																			Expect(result).To(Equal(resultRequeue))
+																		})
+
+																		It("Should return requeue if succeeded to create the SSH Service for the internal builder", func() {
+																			// given
+																			serviceRepository.EXPECT().Create(requestContext, workerSSHService).Return(nil)
+																			// when
+																			result, err := reconciler.Reconcile(requestContext, request)
+																			// then
+																			Expect(err).To(BeNil())
+																			Expect(result).To(Equal(resultQuickRequeue))
+																		})
+																	})
+
+																	Context("SSH Service for the internal builder exists", func() {
+																		const (
+																			workerCertificateNameFormat = "worker-%s-cert"
+																		)
+
+																		var (
+																			internalBuilderCertificateName = fmt.Sprintf(workerCertificateNameFormat, internalBuilderName)
+																			internalBuilderCertificate     *certmanagerv1.Certificate
+																		)
+
+																		BeforeEach(func() {
+																			serviceRepository.EXPECT().Read(requestContext, workerSSHServiceName, operatorNamespace).Return(workerSSHService, nil)
+																			internalBuilderCertificate = &certmanagerv1.Certificate{
+																				ObjectMeta: metav1.ObjectMeta{
```

Comment:
Maybe is can be part of the beforeEach section and be reused for line 1563

Reply:
Not sure what you mean. First, this is a BeforeEach and second this is a certificate and the code in line 1563 is for a secret

Reply:
the ObjectMeta looks the same

Reply:
But, one is for the Certificate and the other is for the Secret. It's just that we configure the certificate to create a secret with the same name. I don't think consolidating it will make the code more readable.

Reply:
OK :+1: